google-cloud-examples/src/main/java/com/google/cloud/examples/bigtable/HelloWorld.java

```diff
@@ -0,0 +1,204 @@
+                .setCell(columnFamily, columnQualifier, greetings[i]);
+        dataClient.mutateRow(rowMutation);
+        System.out.println(greetings[i]);
+      }
+    } catch (ApiException e) {
+      System.err.println("Exception while writing to table: " + e.getMessage());
+    }
+    // [END writing_rows]
+  }
+
+  public void readSingleRow() {
+    // [START reading_a_row]
+    try {
+      System.out.println("\nReading a single row by row key");
+      Row row = dataClient.readRow(tableId, rowKeyPrefix + 0);
+      System.out.println("Row: " + row.getKey().toStringUtf8());
+      for (RowCell cell : row.getCells()) {
+        System.out.printf(
+            "Family: %s    Qualifier: %s    Value: %s%n",
+            cell.getFamily(), cell.getQualifier().toStringUtf8(), cell.getValue().toStringUtf8());
+      }
+    } catch (ApiException e) {
+      System.err.println("Exception while reading a single row: " + e.getMessage());
+    }
+    // [END reading_a_row]
+  }
+
+  public void readTable() {
+    // [START scanning_all_rows]
+    try {
+      System.out.println("\n================= Reading the entire table =================");
+      Query query = Query.create(tableId);
+      ServerStream<Row> rowStream = dataClient.readRows(query);
+      for (Row r : rowStream) {
+        System.out.println("Row Key: " + r.getKey().toStringUtf8());
+        for (RowCell cell : r.getCells()) {
+          System.out.printf(
+              "Family: %s    Qualifier: %s    Value: %s%n",
+              cell.getFamily(), cell.getQualifier().toStringUtf8(), cell.getValue().toStringUtf8());
+        }
+      }
+    } catch (ApiException e) {
+      System.err.println("Exception while reading table: " + e.getMessage());
+    }
+    // [END scanning_all_rows]
+  }
+
+  public void deleteTable() {
+    // [START deleting_a_table]
+    System.out.println("\nDeleting table");
+    try {
+      adminClient.deleteTable(tableId);
+      System.out.printf("Table %s deleted successfully%n", tableId);
+    } catch (ApiException e) {
+      System.err.println("Exception while deleting table: " + e.getMessage());
+    }
+    // [END deleting_a_table]
+  }
+
+  private static String generateTableId() {
+    return String.format(
+        "%s-%016x-%d", tablePrefix, System.currentTimeMillis(), new Random().nextLong());
+  }
+
+  public void garbageCollect() {
+    Pattern timestampPattern = Pattern.compile(tablePrefix + "-([0-9]+)");
+    for (String tableId : adminClient.listTables()) {
+      Matcher matcher = timestampPattern.matcher(tableId);
+      if (!matcher.matches()) {
+        continue;
+      }
+      String timestampStr = matcher.group(1);
+      long timestamp = Long.parseLong(timestampStr);
+      if (System.currentTimeMillis() - timestamp < TimeUnit.MINUTES.toMillis(15)) {
+        continue;
+      }
+      System.out.println("Garbage collecting orphaned table: " + tableId);
+      adminClient.deleteTable(tableId);
+    }
+  }
+}
```

Comment:
Please move generateTableId & garbageCollect to the unit test. No reason to burden the customer with this complexity. It's only needed for automated tests. Have main() hardcode the tableId and have the unit test generate a unique one and garbage collect it